server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesUpdateProcessor.java

```diff
@@ -71,10 +73,14 @@ private void handleUpdate(String mailboxId, MailboxUpdateRequest updateRequest,
         try {
             validateMailboxName(updateRequest, mailboxSession);
             Mailbox mailbox = getMailbox(mailboxId, mailboxSession);
+            checkRole(mailbox.getRole());
```

Comment:
Here we have adopted this solution because we didn't want to multiply the number of lines by adding a lot of 'if'.
Moreover, using exceptions in this context allows us  to be in a functional logic.